weed/s3api/auth_credentials.go

```diff
@@ -364,14 +385,25 @@ func (iam *IdentityAccessManagement) authRequest(r *http.Request, action Action)
 		r.Header.Set(s3_constants.AmzAuthType, "Jwt")
 		return identity, s3err.ErrNotImplemented
 	case authTypeAnonymous:
-		authType = "Anonymous"
-		if identity, found = iam.lookupAnonymous(); !found {
-			r.Header.Set(s3_constants.AmzAuthType, authType)
-			return identity, s3err.ErrAccessDenied
+		if supportAcl && br != nil {
+			bucket, _ := s3_constants.GetBucketAndObject(r)
+			bucketMetadata, errorCode := br.GetBucketMetadata(bucket)
+			if errorCode != s3err.ErrNone {
+				return nil, errorCode
+			}
+			if bucketMetadata.ObjectOwnership != s3_constants.OwnershipBucketOwnerEnforced {
+				return iam.identityAnonymous, s3err.ErrNone
+			}
 		}
+		authType = "Anonymous"
+		identity = iam.identityAnonymous
```

Comment:
Here's the translation:

"If ACL is supported, the bucket's ownership configuration must be checked to decide whether to enforce ACL permissions:
1. If the ownership is set to `BucketOwnerEnforced`, then follow the previous logic (ignore ACL).
2. Otherwise, both the bucket ACL and object ACL need to be checked to determine if anonymous access is allowed."